py/desispec/io/spectra.py

```diff
@@ -241,63 +271,123 @@ def read_spectra(infile, single=False):
     for h in range(1, nhdu):
         name = hdus[h].read_header()["EXTNAME"]
         if name == "FIBERMAP":
-            fmap = encode_table(Table(hdus[h].read(), copy=True).as_array())
+            if not skip_hdu["FIBERMAP"]:
+                fmap = encode_table(
+                    Table(
+                        hdus[h].read(rows=rows, columns=select_columns["FIBERMAP"]),
+                        copy=True,
+                    ).as_array()
+                )
         elif name == "EXP_FIBERMAP":
-            expfmap = encode_table(Table(hdus[h].read(), copy=True).as_array())
+            if not skip_hdu["EXP_FIBERMAP"]:
+                expfmap = encode_table(
+                    Table(
+                        hdus[h].read(rows=rows, columns=select_columns["EXP_FIBERMAP"]),
+                        copy=True,
+                    ).as_array()
+                )
         elif name == "SCORES":
-            scores = encode_table(Table(hdus[h].read(), copy=True).as_array())
-        elif name == 'EXTRA_CATALOG':
-            extra_catalog = encode_table(Table(hdus[h].read(), copy=True).as_array())
+            if not skip_hdu["SCORES"]:
+                scores = encode_table(
+                    Table(
+                        hdus[h].read(rows=rows, columns=select_columns["SCORES"]),
+                        copy=True,
+                    ).as_array()
+                )
+        elif name == "EXTRA_CATALOG":
+            if not skip_hdu["EXTRA_CATALOG"]:
+                extra_catalog = encode_table(
+                    Table(
+                        hdus[h].read(
+                            rows=rows, columns=select_columns["EXTRA_CATALOG"]
+                        ),
+                        copy=True,
+                    ).as_array()
+                )
         else:
             # Find the band based on the name
             mat = re.match(r"(.*)_(.*)", name)
             if mat is None:
-                raise RuntimeError("FITS extension name {} does not contain the band".format(name))
+                raise RuntimeError(
+                    "FITS extension name {} does not contain the band".format(name)
+                )
             band = mat.group(1).lower()
             type = mat.group(2)
             if band not in bands:
                 bands.append(band)
             if type == "WAVELENGTH":
                 if wave is None:
                     wave = {}
-                #- Note: keep original float64 resolution for wavelength
+                # - Note: keep original float64 resolution for wavelength
                 wave[band] = native_endian(hdus[h].read())
             elif type == "FLUX":
                 if flux is None:
                     flux = {}
-                flux[band] = native_endian(hdus[h].read().astype(ftype))
+                if np.any(rows == None):
```

Comment:
I think slightly more standard notation would be `if rows is None:` which works whether rows is None or an array.  The current implementation implies to me that it is checking if any elements of a rows array are None, which I think never happens, and it took me a moment to understand that you were checking if rows itself was None.